controllers/remove_process_groups.go

```diff
@@ -138,108 +141,112 @@ func (u removeProcessGroups) reconcile(ctx context.Context, r *FoundationDBClust
 	return nil
 }
 
-func removeProcessGroup(ctx context.Context, r *FoundationDBClusterReconciler, cluster *fdbv1beta2.FoundationDBCluster, processGroupID fdbv1beta2.ProcessGroupID) error {
-	listOptions := internal.GetSinglePodListOptions(cluster, processGroupID)
-	pods, err := r.PodLifecycleManager.GetPods(ctx, r, cluster, listOptions...)
-	if err != nil {
-		return err
-	}
+func removeProcessGroup(ctx context.Context, r *FoundationDBClusterReconciler, cluster *fdbv1beta2.FoundationDBCluster, processGroup *fdbv1beta2.ProcessGroupStatus) error {
+	podName := processGroup.GetPodName(cluster)
+	var deletionError error
 
-	if len(pods) == 1 && pods[0].DeletionTimestamp.IsZero() {
-		err = r.PodLifecycleManager.DeletePod(ctx, r, pods[0])
+	pod, err := r.PodLifecycleManager.GetPod(ctx, r, cluster, podName)
+	if err == nil && pod.DeletionTimestamp.IsZero() {
+		deletionError = r.PodLifecycleManager.DeletePod(ctx, r, pod)
 		if err != nil {
```

Comment:
We are checking `err==nil` in the outer if statement so this check will be useless.

Reply:
Should be `deletionError`